### ISSUE TITLE
Build failure with rustc 1.16.0-dev (0f8a29647 2017-01-28)

### ISSUE THREAD

**Issue by ishitatsuyuki** [Manishearth/rust-clippy#1485]
Disclaimer: I'm using a hand built rustc, but it shouldn't be the issue anyway.

Full log available at https://paste.rs/dbZ

**Comment by sinkuu** [comment]
#1475

**Comment by oli-obk** [comment]
we don't have a nightly yet: http://rusty-dash.com/nightlies so I'd rather not fix this until people have a way to update.

**Comment by mcarton** [comment]
But if anyone wants to make a PR to be ready when that moment comes, feel free to do so!

**Comment by jplatte** [comment]
This landed in `rustc 1.16.0-nightly (24055d0f2 2017-01-31)`, right?

**Comment by iliekturtles** [comment]
For anyone else who ran into this issue the 2017-01-31 nightly has #1475, 2017-01-26 will compile the current clippy code as of the time of this post:

```
$ rustup install nightly-2017-01-26
$ cargo +nightly-2017-01-26 build --features "clippy"
```

**Comment by choleraehyq** [comment]
Also rust-lang/rust#39230 and rust-lang/rust#39000 are breakage.

**Comment by oli-obk** [comment]
I published 0.0.113
